pkg/engine/lifecycletest/target_test.go

```diff
@@ -3405,3 +3405,89 @@ func TestUntargetedDependencyChainsArePreserved(t *testing.T) {
 		})
 	})
 }
+
+func TestUntargetedProviderChange(t *testing.T) {
+	// This test is a regression test for https://github.com/pulumi/pulumi/issues/14254. This was "fixed" by
+	// https://github.com/pulumi/pulumi/pull/15716 but we didn't notice. This test is to ensure that the issue
+	// stays fixed because we _almost_ regressed it in https://github.com/pulumi/pulumi/pull/17245.
```

Comment:
Possible to add a one-liner on what the test is? It's sorta possible to infer it from the comments below but might be useful to have a summary. Also I'd just move this to be the doc-comment for the function.

Reply:
Sure

Reply:
Comment read ok now?

Reply:
Awesome :ok_hand: Queued :crossed_fingers: